### PR TITLE
Cleanup Inventory app webpack config

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

A bunch of cruft was left in webpack configs from the webpack-4 to v5 migration, old fixed webpack bugs, fixed sissues in server code used by client etc.

This cleans up the config a bit.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:21]: As far as I can tell the webpack config docs ( https://webpack.js.org/configuration/resolve/#resolveextensionalias ) don't explain how extensionAlias and extensions relate to each other, but by testing it empirically, it appears we only need to list the output of the aliasing in the list of extensions, so I have removed the unneeded entries.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:44]: these bugs seem to be fixed, and removing this item doesn't seem to cause any issues as far as I can tell.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:45]: The docs say to use library.type now, so this was updated.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:51]: I believe a mistake in server code consumed by client used to cause this to be needed, but that was fixed and it works fine without it.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:55]: I still don't have a great understanding of whats going on here, but I think there is a separate config for static file content watching which the v5 migration guide incorrectly pointed to (or I just misunderstood it at the time). Regardless this API is documented and works fine, and we have finished the migration and that didn't impact this, so I see no value to these comments any more.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [inline examples/data-objects/inventory-app/webpack.config.cjs:60]: I don't find having 3 webpack files useful, especially given that if you run `webpack --config webpack.prod.config` it doesn't work, since those files weren't actually webpack configs, but part of this one. I think its simpler an ess likley to confuse people to just inline them here.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (jason-ha) [reply examples/data-objects/inventory-app/webpack.config.cjs:21]: That has also been my understanding. Since we are being explicit with paths we may not need this at all. The default is [.js, .json, .wasm]. So, this makes it more restrictive. The thing I don't know is if we need .cjs / .mjs for anything we depend on that doesn't have an explicit extension. We might only find that out in a package that encounters such a dependency.
Setting `enforceExtension: true` would be good.
The above `extensionAlias` setting has been confirmed incorrect. We currently think the correct general setting to be:
```json
			extensionAlias: {
				".cjs": [".cts", ".cjs"],
				".js": [".ts", ".tsx", ".js"],
				".mjs": [".mts", ".mjs"],
			},
```

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (jason-ha) [inline examples/data-objects/inventory-app/webpack.config.cjs:36]:
```json
						test: /\.m?js$/,
						use: [require.resolve("source-map-loader")],
```
is also likely incorrect. I don't know about source-map-loader, but if `m?` is important, then probably `[cm]?` is correct.

**Review by jason-ha — COMMENTED** [review]
Cool - I especially like to see @types/node dependency get cleaned up.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (ChumpChief) [reply examples/data-objects/inventory-app/webpack.config.cjs:55]: Do we know why node_modules is ignored?  Seems we would want to watch changes to node_modules as well (particularly being a monorepo)?

**Review by ChumpChief — APPROVED** [review]
LGTM assuming the extensions/aliasing is correct (I see Jason has some comments that probably need resolving).  Please do also file a work item to apply these changes uniformly across the repo.  @markfields this could be a decent tech debt work item.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [reply examples/data-objects/inventory-app/webpack.config.cjs:21]: I can confirm we work without extensions, as well as with random garbage as the extensions list (so its not that default works, but that anything works). I think since we can build with enforceExtension, then `extensions` is never used.

Thus I'll remove `extensions` and add `enforceExtension: true`.

For extensionAlias I'd do as you recommend, but I'll omit explicit cjs support: we don't require it and I suspect its mainly used when mjs/esm is also available which we would prefer to use.

Comment (CraigMacomber) [reply examples/data-objects/inventory-app/webpack.config.cjs:21]: Annoyingly, webpack's hot module reloading code fails with enforeceExtensions:

```
ERROR in ../../../node_modules/.pnpm/webpack@5.89.0_webpack-cli@4.10.0/node_modules/webpack/hot/emitter.js 1:19-36
Module not found: Error: Can't resolve 'events' in '/home/craig/Work/FluidFramework/node_modules/.pnpm/webpack@5.89.0_webpack-cli@4.10.0/node_modules/webpack/hot'
The request 'events' failed to resolve only because 'resolve.enforceExtension' was specified.
```

I can confirm we work without extensions. As omitting it gets us a default with wasm support which might be nice some day, and maybe some third party lib will use json too, seems good to use the default there I guess.

For extensionAlias I'd do as you recommend, but I'll omit explicit cjs support: we don't require it and I suspect its mainly used when mjs/esm is also available which we would prefer to use.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (tylerbutler) [reply examples/data-objects/inventory-app/webpack.config.cjs:60]: Fewer files === awesome. <3

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [reply examples/data-objects/inventory-app/webpack.config.cjs:36]: I got this from @Abe27342 . Not that I have extensionAlias explicitly for mjs and js, maybe this makes sense.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (CraigMacomber) [reply examples/data-objects/inventory-app/webpack.config.cjs:55]: I think its mostly due to possible issues trying to watch hundreds of thousands of files. Oddly I don't have any issues if I remove that, but I do have issues with jest trying to watch files when running tests, so maybe the jest setup is just extra bad somehow? Sadly the file watching APIs are OS and even somewhat filesystem dependent, so its hard to know if removing this ignore will cause issues for some people.

Note that even if we did watch node modules, if changing other packages you would still need to build those packages explicitly since that wouldn't be watching their source.

**Review by clarenceli-msft — APPROVED** [review]

**Comment by CraigMacomber** [comment]
Its likely further cleanup is practical, but I'll merge this as is since its progress and has approval.

File: examples/data-objects/inventory-app/webpack.config.cjs
Comment (clarenceli-msft) [reply examples/data-objects/inventory-app/webpack.config.cjs]: It was merged but I just thought of one question, do we need `".mts"` as a mapping target? I did not see any instances of `.mts` files being used in our repo. @jason-ha

**Pull request merged**
